doc/nrf/ug_creating_matter_accessory.rst

```diff
@@ -0,0 +1,274 @@
+
+To model behaviour of a sensor you need to add three new events:
+
+#. Sensor activation.
+#. Sensor deactivation.
+#. Sensor measurement.
+
+| Add new events to ``EventType`` in :file:`src/app_event.h`, for example:
+.. code-block:: C++
+        
+        enum EventType : uint8_t { FunctionPress, FunctionRelease, FunctionTimer, SensorActivate, SensorDeactivate, SensorMeasure};
+
+Timers
+------
+
+| One way to make sure that sensor is making measurements in desired time points is to use a Zephyr timer to post ``SensorMeasure`` events.
+| In Template sample such a timer is being used in order to countdown 6 seconds of **Button 1** being pressed to initiate factory reset.
+
+New timer for a measurement event can be added in a similar way:
+
+.. code-block:: C++
+
+        k_timer sSensorTimer;        
+
+        void SensorTimerHandler(k_timer *timer)
+        {
+                GetAppTask().PostEvent(AppEvent{ AppEvent::SensorMeasure });
+        }
+        
+        void StartSensorTimer(uint32_t aTimeoutMs)
+        {
+	        k_timer_start(&sSensorTimer, K_MSEC(aTimeoutMs), K_NO_WAIT);
+        }
+
+        int AppTask::Init()
+        {
+                /*
+                 .      Original content       
+                 */
+
+                k_timer_init(&sSensorTimer, &SensorTimerHandler, nullptr);
+                k_timer_user_data_set(&sSensorTimer, this);
+
+                StartSensorTimer(500);
+                return 0;
+        }
+
+        
+Now every time ``StartSensorTimer(TIMEOUT);`` is called, then after ``TIMEOUT`` milliseconds ``SensorMeasure`` event will be added to the event queue.
+
+
+Event Handlers:
+---------------
+
+| When an event is dequeued the event handler function is called.
+| New events have been added, so corresponding handlers should be implemented, for example (``sensorActive`` will be application variable indicating whether sensor is active):
+.. code-block:: C++
+
+        void AppTask::SensorActivateHandler()
+        {
+	        sensorActive = true;
+        }
+
+        void AppTask::SensorDeactivateHandler()
+        {
+	        sensorActive = false;
+        }
+
+        void AppTask::SensorMeasureHandler()
+        {
+	        if(sensorActive)
+		        emberAfTemperatureMeasurementClusterSetMeasuredValueCallback(1, int16_t(rand()));
+
+	        StartSensorTimer(500);
+        }
+
+        
+When SensorMeasureHandler is called then TemperatureMeasurement cluster's attribute is updated by calling a function from :file:`$MATTER_ROOT/src/app/clusters/temperature-measurement-server.cpp`.
+Because new timer is started every time last timer expires, then SensorMeasureHandler will be called every half a second and will update a value if a sensor is active.
+In this example value is updated randomly, but in a real sensor application it would be updated with a value obtained from external measurement.  
+
+
+Don't forget to declare these handler functions in :file:`src/app_task.h` for a proper build.
```

Comment:
Would calling `StartSensorTimer` directly in the `DispatchEvent` method remove the need for these declarations? Maybe it's worth if they're one-liners?